py3-cassandra-medusa.yaml

```diff
@@ -96,5 +103,5 @@ test:
       packages:
         - python-3.11-dev
   pipeline:
-    - runs: |
-        medusa --version
+    - runs: medusa --version
+    - runs: /usr/share/medusa/.venv/bin/python3 -m medusa.service.grpc.server server.py 2>&1 | grep "/etc/medusa/medusa.ini"
```

Comment:
this fails to capture error if stdout happened to have `/etc/medusa/medusa.ini` in it.

for example, the left side of the pipe could be this program and you'd pass:

```
#!/bin/sh
echo PANIC PANIC PANIC no /etc/medusa.ini 
exit 1
```

you can avoid ignoring 'python3' exiting non-zero like this:

```yaml
runs: |
    fail() { echo "$@" 1>&2; exit 1; }
    out=$(/usr/share/medusa/.venv/bin/python3 -m medusa.service.grpc.server 2>&1) ||
        fail "medusa.service.grpc.server exited $?: $out"
    echo "$out" | grep -q /etc/medusa/medusa.ini || fail "medusa.service.grpc.server output did not contain 'medusa.ini'"
```

These are just shell footguns that we need to get rid of.

we need a library of test tools to work with so that you could write something like this:

```yaml:
 - runs: |
      assert --exit=0 --output-contains=/etc/medusa/medusa.ini \
           "medusa.service.grpc.server output contains medusa.ini" -- \
           /usr/share/medusa/.venv/bin/python3 -m medusa.service.grpc.server server.py
```

Reply:
Thanks! Revisited your test snippet to following:
```yaml
- runs: |
         set +e
         fail() { echo "$@" 1>&2; exit 1; }
         out=$(/usr/share/medusa/.venv/bin/python3 -m medusa.service.grpc.server 2>&1)
         status=$?
         echo "$out" | grep -q '/etc/medusa/medusa.ini' || fail "medusa.service.grpc.server output did not contain expected 'medusa.ini' message. Exit status $status: $out"
         echo "medusa.service.grpc.server exited with expected error message"
```